scrapy/exporters.py

```diff
@@ -144,6 +144,7 @@ def __init__(self, file, **kwargs):
         if not self.encoding:
             self.encoding = 'utf-8'
         self.xg = XMLGenerator(file, encoding=self.encoding)
+        self.file = self.xg
```

Comment:
Oh wait this is wrong, this will generate a warning everytime XML exporter is used. Will fix this.